### PR TITLE
check status before DatabricksSubmitRunOperator & DatabricksSubmitRunOperator executes in deferrable mode

### PR DESCRIPTION
**Pull request by vatsrahul1001**
This PR addresses checking the status before deferring the task to the triggerer.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: tests/providers/databricks/operators/test_databricks.py [hunk 1]
```diff
@@ -982,6 +982,68 @@ def test_execute_complete_incorrect_event_validation_failure(self):
         with pytest.raises(AirflowException):
             op.execute_complete(context=None, event=event)
 
+    #
```
Comment (Lee-W) [inline tests/providers/databricks/operators/test_databricks.py]:
```suggestion
```

File: airflow/providers/databricks/operators/databricks.py [hunk 2]
```diff
@@ -132,18 +132,25 @@ def _handle_deferrable_databricks_operator_execution(operator, hook, log, contex
...
+                    retry_delay=operator.databricks_retry_delay,
+                    retry_args=operator.databricks_retry_args,
+                    run_page_url=run_page_url,
+                ),
+                method_name=DEFER_METHOD_NAME,
+            )
+        else:
+            if run_state.is_successful:
+                log.info("%s completed successfully.", operator.task_id)
+                return
```
Comment (Lee-W) [inline airflow/providers/databricks/operators/databricks.py]:
```suggestion
```

**Review by Lee-W — APPROVED** [review]
Left some nitpick. other looks good to me 🙂

**Comment by Lee-W** [comment]
It seems there are some failed tests. Should we rebase and see whether it fixed the issue. If not, I think we'll need to take a deeper look

**Review by pankajastro — APPROVED** [review]
LGTM

**Pull request merged**